File: vdp-php/tests/visacheckout/GetPaymentDataTest.php
<?php

namespace Vdp;

class GetPaymentDataTest extends \PHPUnit_Framework_TestCase {
	
	public function setUp() {
		$this->conf = parse_ini_file ( "configuration.ini", true );
		$this->abstractVisaAPIClient = new AbstractVisaAPIClient;
	}
	
	public function testGetPaymentInfo() {
		$baseUrl = "wallet-services-web/";
		$resourcePath = "payment/data/{callId}";
		$resourcePath = str_replace("{callId}",$this->conf ['VDP'] ['checkoutCallId'],$resourcePath);
		$queryString = "apikey=".$this->conf ['VDP'] ['apiKey'];
		$statusCode = $this->abstractVisaAPIClient->doXPayTokenCall ( 'get', $baseUrl, $resourcePath, $queryString, 'Get Payment Information Test', '');
		$this->assertEquals($statusCode, "200");
	}
}

File: vdp-c-sharp/vdp-c-sharp/VDPApiTest/visacheckout/GetPaymentDataTest.cs
using System.Configuration;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Vdp
{
    [TestClass]
    public class GetPaymentDataTest
    {
        private string apikey;
        private AbstractVisaAPIClient abstractVisaAPIClient;

        public GetPaymentDataTest()
        {
            abstractVisaAPIClient = new AbstractVisaAPIClient();
            apikey = ConfigurationManager.AppSettings["apiKey"];
        }

        [TestMethod]
        public void TestGetPaymentInfo()
        {
            string baseUri = "wallet-services-web/";
            string resourcePath = "payment/data/{callId}";
            resourcePath = resourcePath.Replace("{callId}", ConfigurationManager.AppSettings["checkoutCallId"]);
            string queryString = "apikey=" + apikey;
            string status = abstractVisaAPIClient.DoXPayTokenCall(baseUri, resourcePath, queryString, "GET", "Get Payment Information Test", "");
            Assert.AreEqual(status, "OK");
        }
    }
}
